src/controllers/mainController.ts

```diff
@@ -336,6 +337,24 @@ export default class MainController implements vscode.Disposable {
                 providerInstance,
             );
 
+            const self = this;
+            const uriHandler: vscode.UriHandler = {
+                handleUri(uri: vscode.Uri): vscode.ProviderResult<void> {
+                    if (self.isExperimentalEnabled) {
```

Comment:
Why put the `isExperimentalEnabled` check inside the handler, instead of outside like some of the other registrations?  I think inside makes sense if we have two different implementations instead of swapping between no implementation and something.